config.go

```diff
@@ -916,44 +917,29 @@ func loadConfig() (*config, []string, error) {
 		}
 	}
 
-	return &cfg, remainingArgs, nil
-}
-
-// findSampleConfigPathWindows looks in %PROGRAMFILES% for the sample
-// configuration file. If it finds it, it returns it. Otherwise, an error
-// is returned.
-func findSampleConfigPathWindows() (string, error) {
-	winProgramFiles := os.Getenv("PROGRAMFILES")
-	winPFConfig := filepath.Join(winProgramFiles, "Decred", "Dcrd",
-		sampleConfigFilename)
-	_, err := os.Stat(winPFConfig)
-	if err == nil {
-		return winPFConfig, nil
-	}
-	if !os.IsNotExist(err) {
-		return "", err
+	// Warn about missing config file only after all other configuration is
+	// done.  This prevents the warning on help messages and invalid
+	// options.  Note this should go directly before the return.
+	if configFileError != nil {
+		dcrdLog.Warnf("%v", configFileError)
 	}
 
-	return "", errors.New("could not find sample configuration to make new " +
-		"automated configuration file from")
+	return &cfg, remainingArgs, nil
 }
 
-// createDefaultConfigFile copies the file sample-dcrd.conf to the given
-// destination path, and populates it with some randomly generated RPC username
-// and password. It will only do this if the operating system is Windows.
-func createDefaultConfigFile() error {
-	if runtime.GOOS != "windows" {
-		return nil
-	}
+// createDefaultConfig copies the file sample-dcrd.conf to the given destination path,
+// and populates it with some randomly generated RPC username and password.
+func createDefaultConfigFile(destinationPath string) error {
+	// Create the destination directory if it does not exists
+	os.MkdirAll(filepath.Dir(destinationPath), 0700)
 
-	sampleConfigPath, err := findSampleConfigPathWindows()
-	if err != nil {
-		return err
-	}
+	// We get the sample config file path, which is in the same directory as this file.
```

Comment:
This is reverting a fix to use the installation location to find the default sample config, going back to the old broken behavior where it copied the file out of the Go source tree.  Remember all the issues I had when testing autoconf on a clean MSI installation?  It was because of this.